.github/workflow.templates/on-push.yml

```diff
@@ -2,54 +2,18 @@
 #@ load("rush.lib.yml", "rush_add_path")
 #@ load("rush.lib.yml", "rush_install")
 #@ load("rush.lib.yml", "rush_build")
+#@ load("test-job.lib.yml", "test_job")
 
 name: Cheks
 
 "on":
   - push
 
 jobs:
-  js-search-test:
-    name: JS-Search tests
-    runs-on: ubuntu-latest
-    timeout-minutes: 10
-    permissions:
-      id-token: write
-      contents: read
-      checks: write
-    steps:
-      - uses: actions/checkout@v2
-      - uses: actions/setup-node@v2
-        with:
-          node-version: "16"
-      -  #@ cache_node()
-      -  #@ rush_add_path()
-      -  #@ rush_install()
-      - name: Run tests
-        run: cd pkg/js-search/ && rushx test:ci
-      - name: Test Report
-        uses: dorny/test-reporter@v1
-        if: success() || failure()
-        with:
-          name: JS-Search Tests
-          path: pkg/js-search/junit.xml
-          reporter: jest-junit
-      - name: Build js-search
-        run: cd pkg/js-search/ && rushx build
-  ui-test:
-    name: UI tests
-    runs-on: ubuntu-latest
-    steps:
-      - uses: actions/checkout@v2
-      - uses: actions/setup-node@v2
-        with:
-          node-version: "16"
-      -  #@ cache_node()
-      -  #@ rush_add_path()
-      -  #@ rush_install()
-      -  #@ rush_build()
-      - name: Run tests
-        run: cd pkg/ui && rushx test:ci
+  js-search-test: #@ test_job("JS-Search Tests", "pkg/js-search", "jest-junit")
```

Comment:
Very small thing, and debatable. But I'd like to point out that this can also be expressed by providing parameter names for improved readability:
```
  js-search-test: #@ test_job("JS-Search Tests", workdir="pkg/js-search", reporter="jest-junit")
```